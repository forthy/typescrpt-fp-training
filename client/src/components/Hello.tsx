import * as React from "react";

export interface HelloProps { compiler: string; framework: string; }

export class Hello extends React.Component<HelloProps, {}> {
    constructor(props: HelloProps) {
        super(props);
    }

    render() {
        return (
            <div data-testid="hello">
                <h1>Hello from {this.props.compiler} and {this.props.framework}!</h1>
            </div>
        );
    }
}