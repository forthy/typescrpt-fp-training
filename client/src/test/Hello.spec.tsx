import React from "react";
import renderer from 'react-test-renderer';
import { Hello } from "../components/Hello";

describe("Hello testing", () => {
    test("test01", () => {
        const component = renderer.create(<Hello compiler="mycompiler" framework="myframework" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})