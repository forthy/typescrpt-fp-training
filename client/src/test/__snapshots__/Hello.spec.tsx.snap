// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello testing test01 1`] = `
<div
  data-testid="hello"
>
  <h1>
    Hello from 
    mycompiler
     and 
    myframework
    !
  </h1>
</div>
`;
